### PR TITLE
DocFX: Multiple version support

### PR DESCRIPTION
**Pull request by MazeXP**
This PR customizes the build workflow to publish multiple versions of
the documentation.

It is setup to always keep the `main` branch and the latest 5 tags (variable `KEEP_TAG_VERSIONS`).
The `index.html` is edited on every build to redirect to the latest tag.

## Remark
Before merging into `main` branch the `gh-pages` branch must be cleaned up, to consist only of following files:

#### - `index.html`
```html
<html>
  <head>
    <meta http-equiv="refresh" content="0; url=./main" />
  </head>
  <body></body>
</html>
```

### CODE REVIEW

File: .github/workflows/dotnet.yml [hunk 1]
```diff
@@ -34,34 +39,89 @@ jobs:
...
         with:
           name: docfx-site
-          path: site
+          path: site/${{ github.ref_name }}
+
+      - name: Manage versions
+        shell: bash
+        working-directory: site
+        run: |
+          # Read versions
```
Comment (Nihlus) [inline .github/workflows/dotnet.yml:95]: This might be better suited as a separate script file.

Comment (MazeXP) [reply .github/workflows/dotnet.yml:95]: Any recommended place?
How about docfx/scripts or .github/scripts?
(Would have to rework the job to also checkout the branch in the `docs` job)

Comment (Nihlus) [reply .github/workflows/dotnet.yml:95]: docfx/scripts sounds good to me.

**Comment by MazeXP** [comment]
> Is there a way to avoid having to do the manual work on the gh-pages branch? Having it sort out the details itself would be far preferable.

If me move the part to a script it would also be possible to have the index.html and .nojekyll as a template inside an `docfx/templates` directory.
In addition we could have an aggressive cleanup script that does every file in the root and every folder that does not match either `main` or the pattern `YYYY.XX`.

**Comment by Nihlus** [comment]
Sounds good!

**Pull request merged**